brackend/endpoints/tournaments/repositories/TournamentRepository.py

```diff
@@ -1,14 +1,46 @@
 from abc import ABC
-from sqlalchemy.orm import Session
+from sqlalchemy.orm import Session, subqueryload
 from brackend.db.models import (
     EngineGetter,
     Tournament,
+    UserTournament,
+    User,
 )
+from brackend.db.enums import UserRole
 
 
 class TournamentRepository(ABC):
     engine = EngineGetter.get_or_create_engine()
 
+    @classmethod
+    def get_by_id(cls, t_id):
+        with Session(cls.engine) as session:
+            session.expire_on_commit = False
+            tournament = session.query(Tournament) \
+                .options(subqueryload(Tournament.brackets)) \
+                .filter(Tournament.id == t_id) \
+                .one_or_none()
```

Comment:
Need to do a subquery load here because we're lazily accessing the `brackets` when going `to_json` on the tournament

If you can think of a better way to do this, let me know. Or maybe we could add a "include_brackets" param if we're worried about doing this eager loading, but it really doesn't seem very risky for near or probably long term

Reply:
This solution looks good to me for now!